syft/frameworks/torch/tensors/interpreters/additive_shared.py

```diff
@@ -525,41 +515,13 @@ def sub(self, shares: dict, other):
                 - a constant
         """
 
-        if isinstance(other, int):
-            other = torch.tensor([other], dtype=self.torch_dtype)
-
-        if isinstance(other, (torch.LongTensor, torch.IntTensor)):
-            # if someone passes a torch tensor, we share it and keep the dict
-            other = other.share(
-                *self.child.keys(),
-                field=self.field,
-                dtype=self.dtype,
-                crypto_provider=self.crypto_provider,
-                **no_wrap,
-            ).child
-        elif not isinstance(other, dict):
-            # if someone passes in a constant, we cast it to a tensor, share it and keep the dict
-            other = (
-                torch.tensor([other], dtype=self.torch_dtype)
-                .share(
-                    *self.child.keys(),
-                    field=self.field,
-                    dtype=self.dtype,
-                    crypto_provider=self.crypto_provider,
-                    **no_wrap,
-                )
-                .child
-            )
+        other = self._convert_to_shared_tensor(other)
 
         assert len(shares) == len(other)
 
         # matches each share which needs to be added according
         # to the location of the share
-        new_shares = {}
-        for k, v in shares.items():
-            new_shares[k] = self.modulo(v - other[k])
-
-        return new_shares
+        return {k: self.modulo(v - other[k]) for k, v in shares.items()}
```

Comment:
you should avoid code repetition, because when we need to change this part of code we will need to change it in multiple places instead of one

Reply:
I agree with you. I will change it 😄

Reply:
I dedupkicated these repeated codes using `apply_to_share`